sr_hand_health_report/src/sr_hand_health_report/monotonicity_check.py

```diff
@@ -25,50 +26,77 @@
 
 class MonotonicityCheck(SrHealthReportCheck):
 
-    PASSED_THRESHOLDS = True
+    PASSED_THRESHOLDS = {'is_monotonic': True, 'higher_raw_sensor_value': 4090, 'lower_raw_sensor_value': 5}
 
     def __init__(self, hand_side, fingers_to_test):
+        """
+            Initialize the MonotonicityCheck object
+            @param hand_side: String indicating the side
+            @param fingers_to_test: List of finger prefixes to test
+        """
         super().__init__(hand_side, fingers_to_test)
+        self._name = "Monotonicity"
         self._is_joint_monotonous = True
         self._dict_of_monotonic_joints = {}
         self._publishing_rate = rospy.Rate(50)  # 50 Hz
         self._older_raw_sensor_value = 0
         self._previous_difference = 0
         self._pwm_command = 250
-        self._check_duration = rospy.Duration(7.0)
+        self._check_duration = rospy.Duration(4.0)
         self._first_end_stop_sensor_value = None
         self._second_end_stop_sensor_value = None
+        self._result = {'monotonicity': {}}
 
     def run_check(self):
-        result = {"monotonicity": []}
+        """
+            Runs the check for all fingers
+        """
+        self._result = {'monotonicity': {}}
+        if self._stopped_execution:
+            self._stopped_execution = False
+            return
         rospy.loginfo("Running Monotonicity Check")
+        result = dict(self._result)
+
         self.move_fingers_to_start_position()
         self.switch_controller_mode("effort")
 
+        Finger.PREFIXES = ("FF", 'MF', 'LF', 'RF', 'TH', 'WR')
```

Comment:
Wasn't this already defined?

Reply:
it was defined in sr_hand_health_report_check as PREFIXES = ("FF", 'MF', 'RF', 'LF', 'TH', 'WR'). The order there is different. As the _fingers_to_check list is being initialized based on the order of joint names in /joint_states topic, I needed to apply a different finger order to avoid fingers crashing into each other. But yeah, this made me think so I'll make some other changes to improve that